grove-legacy-routes/proxy.js

```diff
@@ -87,9 +87,15 @@ var provider = (function() {
           // call backend, and pipe clientResponse straight into res
           backend.call(req, reqOptions, null, res);
         },
-        function() {
-          // TODO: might return an error too?
-          four0four.unauthorized(req, res);
+        function(error) {
+          if (error !== 'Unauthorized') {
+            throw error;
+          }
+          if (rule.authed) {
+            four0four.unauthorized(req, res);
+          } else {
+            backend.call(req, reqOptions, null, res);
+          }
```

Comment:
@grtjn wrote:
Does this actually work? I can imagine that ML will persist, and through another 401. It might need to set some default user/pass in the reqOptions to make this work, but that raises the question where to get those safely. Like I mentioned in JIRA, it would be best if ML allowed unauthed access in those cases, but that might be hard with out of the box ML REST api..

Reply:
In this case, the developers were using application-level authentication at the MarkLogic layer. That is the use-case I imagine this fixing. If an alternative authorization is desired, in the Grove Node middle-tier, the proper solution would be to provide an alternative authProvider, I believe. I don't think `{auth: false}` in the proxy whitelist has to handle alternative auth.